control-plane/roles/headscale/templates/headscale.yaml

```diff
@@ -120,9 +122,9 @@ metadata:
     nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
     nginx.ingress.kubernetes.io/ssl-redirect: "true"
 {% endif %}
-{% for key, value in headscale_ingress_annotations.items() %}
-    {{ key }}: "{{ value }}"
-{% endfor %}
+{% if cert_manager_cluster_issuer is defined %}
```

Comment:
Just a map of ingress annotations would be more versatile, the you can freely configure your ingress as a user.